### PR TITLE
Remove use of DictCursor for use_role

### PR DESCRIPTION
**Pull request by sfc-gh-fcampbell**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Query Redaction (an enterprise feature) also redacts column names, which breaks `DictCursor`. Let's remove it from `use_role` and use a column index since the query has a well-defined column order in its output.

### CODE REVIEW

File: src/snowflake/cli/api/sql_execution.py [hunk 1]
```diff
@@ -98,22 +98,15 @@ def use(self, object_type: ObjectType, name: str):
...
-        return role_result["CURRENT_ROLE()"]
+        return self._execute_query(f"select current_role()").fetchone()[0]
 
     @contextmanager
     def use_role(self, new_role: str):
         """
         Switches to a different role for a while, then switches back.
         This is a no-op if the requested role is already active.
         """
-        role_result = self._execute_query(
```
Comment (sfc-gh-fcampbell) [inline src/snowflake/cli/api/sql_execution.py:113]: `_execute_query` ends up calling `_execute_string` anyways, so the behaviour here doesn't change

**Review by sfc-gh-cgorrie — APPROVED** [review]

File: src/snowflake/cli/api/sql_execution.py [hunk 2]
```diff
@@ -98,22 +98,15 @@ def use(self, object_type: ObjectType, name: str):
             )
 
     def current_role(self) -> str:
-        *_, cursor = self._execute_string(
-            "select current_role()", cursor_class=DictCursor
-        )
-        role_result = cursor.fetchone()
-        return role_result["CURRENT_ROLE()"]
+        return self._execute_query(f"select current_role()").fetchone()[0]
```
[the remark this answers is not in the document]
Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/api/sql_execution.py:101]: Good question, we are using DictCursor in many places

[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/api/sql_execution.py:101]: It's affecting other queries, there are many integration tests that fail if this parameter is enabled. I'm gonna make separate PRs for these

**Review by sfc-gh-melnacouzi — APPROVED** [review]

File: src/snowflake/cli/api/sql_execution.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sfc-gh-melnacouzi) [reply src/snowflake/cli/api/sql_execution.py:101]: Do we have though any guarantee about column order? i.e. do we always add column at the end for backward compatibility?

[the remark this answers is not in the document]
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/api/sql_execution.py:101]: for this query there's only one column, others will be handled on a case-by-case basis

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**